src/Components/Aspire.Azure.Messaging.ServiceBus/AspireServiceBusExtensions.cs

```diff
@@ -61,10 +124,112 @@ public static void AddKeyedAzureServiceBusClient(
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureMessagingServiceBusSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{TClient, TOptions}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire:Azure:Messaging:ServiceBus:{name}" section.</remarks>
+    /// <exception cref="InvalidOperationException">Thrown when neither <see cref="AzureMessagingServiceBusSettings.ConnectionString"/> nor <see cref="AzureMessagingServiceBusSettings.FullyQualifiedNamespace"/> is provided.</exception>
+    public static void AddKeyedAzureServiceBusProcessor(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<AzureMessagingServiceBusSettings>? configureSettings = null,
+        Action<IAzureClientBuilder<ServiceBusProcessor, ServiceBusProcessorOptions>>? configureClientBuilder = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        new ServiceBusProcessorComponent().AddClient(builder, DefaultConfigSectionName, configureSettings, configureClientBuilder, connectionName: name, serviceKey: name);
+    }
+
+    private abstract class ServiceBusComponent<TClient, TOptions> : AzureComponent<AzureMessagingServiceBusSettings, TClient, TOptions>
+        where TClient : class
+        where TOptions : class
+    {
+        protected override IHealthCheck CreateHealthCheck(TClient client, AzureMessagingServiceBusSettings settings)
+        {
+            return !string.IsNullOrEmpty(settings.HealthCheckQueueName)
+                ? new AzureServiceBusQueueHealthCheck(new AzureServiceBusQueueHealthCheckOptions(settings.HealthCheckQueueName)
+                {
+                    FullyQualifiedNamespace = settings.FullyQualifiedNamespace,
+                    ConnectionString = settings.ConnectionString,
+                    Credential = settings.Credential
+                })
+                : new AzureServiceBusTopicHealthCheck(new AzureServiceBusTopicHealthCheckOptions(settings.HealthCheckTopicName!)
+                {
+                    FullyQualifiedNamespace = settings.FullyQualifiedNamespace,
+                    ConnectionString = settings.ConnectionString,
+                    Credential = settings.Credential
+                });
+        }
+
+        protected override void BindSettingsToConfiguration(AzureMessagingServiceBusSettings settings, IConfiguration config)
+        {
+            config.Bind(settings);
+        }
+
+        protected override bool GetHealthCheckEnabled(AzureMessagingServiceBusSettings settings)
+            => !string.IsNullOrEmpty(settings.HealthCheckQueueName) || !string.IsNullOrEmpty(settings.HealthCheckTopicName);
```

Comment:
Now that we have a Queue or Topic name, I wonder if we should add the Health Check based on that Queue/Topic.